### PR TITLE
[Dy2Stat] Fix bug: Do not use gast.Subscript to replace gast.Name in when transforming for_enumerate_loop

### PR DESCRIPTION
**Pull request by liym27**
### PR types

Bug fixes
### PR changes

Others
### Describe

As the title

### CODE REVIEW

**Review by yghstill — DISMISSED** [review]

File: python/paddle/fluid/dygraph/dygraph_to_static/utils.py [hunk 1]
```diff
@@ -772,6 +773,20 @@ class NameNodeReplaceTransformer(gast.NodeTransformer):
 
     def __init__(self, root_node, target_name, replace_node):
         assert isinstance(target_name, str)
+
+        # NOTE(liym27):
+        # Use gast.Name to replace gast.Name, otherwise, errors may occur.
+        #
+        # For examples:
+        # If using a gast.Subscript to replace gast.Name, and the original gast.Name
+        # is in the arguments of FunctionDef, an exceptions will be raised.
```
Comment (zhhsplendid) [inline python/paddle/fluid/dygraph/dygraph_to_static/utils.py]: Tiny grammar issue: an exception

**Review by Aurelius84 — DISMISSED** [review]
LGTM

**Review by zhhsplendid — COMMENTED** [review]
You can change the grammar comment or in next PR. LGTM

File: python/paddle/fluid/dygraph/dygraph_to_static/utils.py (hunk 1 above)
Comment (liym27) [reply python/paddle/fluid/dygraph/dygraph_to_static/utils.py]: Thanks, done.

**Review by chenwhql — APPROVED** [review]
LGTM

**Review by Aurelius84 — APPROVED** [review]

**Review by zhhsplendid — APPROVED** [review]
LGTM

**Pull request merged**